### PR TITLE
EZP-32214: Refactored config/packages/overrides/* to load from Extension

### PR DESCRIPTION
**Pull request by webhdx**
| Question           | Answer
| ------------------ | ------------------
| **JIRA issue**     | [EZP-32214](https://jira.ez.no/browse/EZP-32214)
| **Improvement**| yes
| **New feature**    | no
| **Target version** | `master`
| **BC breaks**      | no
| **Tests pass**     | yes
| **Doc needed**     | no

Second attempt at refactoring `config/packages/overrides/*.php` files. Previously it wasn't fully working. It has to be a part of `Extension::load` in order for the data to be ready for compiler passes.

### CODE REVIEW

File: src/EzPlatformCoreBundle/bundle/DependencyInjection/EzPlatformCoreExtension.php [hunk 1]
```diff
@@ -47,4 +51,255 @@ public function prepend(ContainerBuilder $container): void
...
+
+        if (isset($relationships['solr'])) {
+            foreach ($relationships['solr'] as $endpoint) {
+                if ($endpoint['scheme'] !== 'solr') {
+                    continue;
+                }
+
+                $container->setParameter('search_engine', 'solr');
+
+                $container->setParameter('solr_dsn', sprintf('http://%s:%d/%s', $endpoint['host'], $endpoint['port'], 'solr'));
```
Comment (alongosz) [inline src/EzPlatformCoreBundle/bundle/DependencyInjection/EzPlatformCoreExtension.php:267]: As you might remember from previous week, this parameter is slightly different in Commerce due to Solarium coupling config conflict. Is there a plan to overcome this issue? My idea was to unify configuration so the version from Commerce works for OSS & DXP, but haven't investigated it yet.

**Review by alongosz — CHANGES_REQUESTED** [review]

File: src/EzPlatformCoreBundle/bundle/DependencyInjection/EzPlatformCoreExtension.php (hunk 1 above)
Comment (webhdx) [reply src/EzPlatformCoreBundle/bundle/DependencyInjection/EzPlatformCoreExtension.php:267]: Yes this is why a few lines below there is `if` statement handling ezcommerce case: https://github.com/ezsystems/ezplatform-core/pull/27/files/b8ce23f032076d82bb56a049830631e0c44e0d97#diff-f72b922417794d0e5c2291bf6eddcfdf0e1e32a18d00f5ff3e8b46c514baa100R236-R244

**Review by adamwojs — APPROVED** [review]
As we discussed on Slack: would be good to move this code into dedicated platform.sh related package. But for now it's OK.

File: src/EzPlatformCoreBundle/bundle/DependencyInjection/EzPlatformCoreExtension.php (hunk 1 above)
Comment (alongosz) [reply src/EzPlatformCoreBundle/bundle/DependencyInjection/EzPlatformCoreExtension.php:267]: > Yes this is why a few lines below there is `if` statement handling ezcommerce case: https://github.com/ezsystems/ezplatform-core/pull/27/files/b8ce23f032076d82bb56a049830631e0c44e0d97#diff-f72b922417794d0e5c2291bf6eddcfdf0e1e32a18d00f5ff3e8b46c514baa100R236-R244

Uh, difficult to spot in this spaghetti. OK.

**Review by alongosz — APPROVED** [review]
Some refactoring in the future of this code would be much appreciated. I'll try to find some time.

**Review by mnocon — CHANGES_REQUESTED** [review]
First, a small thing: 

1) there's no handling of fastly purge_server (Fastly purge_server is not set:
https://github.com/ezsystems/ezplatform-ee/blob/master/config/packages/overrides/generic.php#L42-L44) which was a nice addition (and could break for people doing an upgrade).

2) SESSION_SAVE_PATH and SESSION_HANDLER_ID seem to have no effect

For:
```
export SESSION_SAVE_PATH=tcp://redis-session:6379?weight=1
export SESSION_HANDLER_ID=ezplatform.core.session.handler.native_redis
```

`php bin/console debug:config FrameworkBundle` still shows current values:

```
    session:
        handler_id: session.handler.native_file
        save_path: /Users/mareknocon/Desktop/Sites/v3_2/var/sessions/dev
```
but it should show (current behaviour):
```
session:
    handler_id: ezplatform.core.session.handler.native_redis
    save_path: 'tcp://redis-session:6379?weight=1
```

I've also tested Commerce on platform.sh:
3) Purge type for Varnish is not configured automatically.

After Platform.sh deployment purge_type still seems to be "local".
If I understand correctly this part of code executes only if the HTTPCACHE_PURGE_TYPE variable is not set: https://github.com/ezsystems/ezplatform-core/pull/27/files#diff-f72b922417794d0e5c2291bf6eddcfdf0e1e32a18d00f5ff3e8b46c514baa100R280
and now we give it a default value.

4) Solr is not configured when deploying on Platform.sh. It is set if I add SEARCH_ENGINE=solr variable to .platform.app.yaml. but without it it does not happen.

I will send you my Platform.sh project on Slack.

**Review by mnocon — APPROVED** [review]
QA approved.

Please remember to remove TMP changes in https://github.com/ezsystems/ezplatform/pull/632 before merging.

**Pull request merged**